test/opengl/update_stage.cpp

```diff
@@ -0,0 +1,61 @@
+// test case provided by Lee Yuguang
+
+
+#include "Halide.h"
+#include <stdio.h>
+#include <chrono>
+
+using namespace Halide;
+
+int main() {
+    // This test must be run with an OpenGL target.
+    const Target target = get_jit_target_from_environment().with_feature(Target::OpenGL);
+
+    // Define the input
+    const int width = 10, height = 12, channels = 2, res_channels = 2;
+    Buffer<float> input(width, height, channels);
+    for (int c = 0; c < input.channels(); c++) {
+        for (int y = 0; y < input.height(); y++) {
+            for (int x = 0; x < input.width(); x++) {
+                input(x, y, c) = float(x + y);
+            }
+        }
+    }
+
+    // Define the algorithm.
+    Var x, y, c;
+    RDom r(3, 5, "r");
+    Func f, g;
+
+    Expr coordx = clamp(x + r, 0, input.width() - 1);
+    f(x, y, c) = cast<uint8_t>(0);
+    f(r.x, y, c) = cast<uint8_t>(11);
+
+
+    // Schedule f and g to compute in separate passes on the GPU.
+    f.bound(c, 0, 2);
+    f.update(0).glsl(r.x, y, c);
```

Comment:
Do you need to call `f.glsl(x, y, c)` to schedule the pure definition on the GPU?